File: WePeiYang/Models/Scheme/SchemeManager.m
//
//  SchemeManager.m
//  WePeiYang
//
//  Created by Qin Yubo on 16/1/24.
//  Copyright © 2016年 Qin Yubo. All rights reserved.
//

#import "SchemeManager.h"
#import "SolaFoundationKit.h"
#import "wpyWebViewController.h"
#import "SVModalWebViewController.h"

@implementation SchemeManager

+ (void)handleSchemeURL:(NSURL *)url {
    NSLog(@"URL scheme: %@", [url scheme]);
    NSLog(@"URL host: %@", [url host]);
    NSLog(@"URL path: %@", [url path]);
    NSLog(@"URL query: %@", [url query]);
    
    if ([[url host] isEqualToString:@"platform"]) {
        if ([[url path] isEqualToString:@"/startapp"]) {
            NSMutableDictionary *queryDic = [[NSMutableDictionary alloc] init];
            NSArray *urlComponents = [[url query] componentsSeparatedByString:@"&"];
            for (NSString *keyValuePair in urlComponents) {
                NSArray *pairComponents = [keyValuePair componentsSeparatedByString:@"="];
                NSString *key = [[pairComponents firstObject] stringByRemovingPercentEncoding];
                NSString *value = [[pairComponents lastObject] stringByRemovingPercentEncoding];
                [queryDic setObject:value forKey:key];
            }
            
            if ([queryDic objectForKey:@"app"] != nil) {
                if ([[queryDic objectForKey:@"app"] isEqualToString:@"h5"]) {
                    if ([queryDic objectForKey:@"url"] != nil) {
                        NSString *url = [queryDic objectForKey:@"url"];
                        SVModalWebViewController *webViewController = [[SVModalWebViewController alloc] initWithAddress:url];
                        [[SolaFoundationKit topViewController] presentViewController:webViewController animated:YES completion:nil];
                    }
                } else if ([[queryDic objectForKey:@"app"] isEqualToString:@"main"]) {
                    
                } else if ([[queryDic objectForKey:@"app"] isEqualToString:@"gpa"]) {
                    
                } else if ([[queryDic objectForKey:@"app"] isEqualToString:@"classtable"]) {
                    
                }
            }
        }
    }

}

@end

File: WePeiYang/Models/Scheme/SchemeManager.swift
//
//  SchemeManager.swift
//  WePeiYang
//
//  Created by Qin Yubo on 16/4/19.
//  Copyright © 2016年 Qin Yubo. All rights reserved.
//

import UIKit

class SchemeManager: NSObject {
    
    class func handleURL(url: NSURL) -> Bool {
        
        if url.scheme == "wepeiyang" {
            if url.host == "platform" {
                
                if url.path == "/startapp" {
                    var queryDic: [String: String] = [:]
                    if let urlComponents = url.query?.componentsSeparatedByString("&") {
                        for keyValuePair in urlComponents {
                            let pairComponents = keyValuePair.componentsSeparatedByString("=")
                            if let key = pairComponents.first?.stringByRemovingPercentEncoding {
                                if let value = pairComponents.last?.stringByRemovingPercentEncoding {
                                    queryDic[key] = value
                                }
                            }
                        }
                    }
                    
                    if queryDic["app"] != nil {
                        switch queryDic["app"]! {
                        case "h5":
                            if queryDic["url"] != nil {
                                let url = queryDic["url"]
                                let webViewController = wpyModalWebViewController(address: url!)
                                SolaFoundationKit.topViewController().presentViewController(webViewController, animated: true, completion: nil)
                            }
                        case "main":
                            break
                        case "gpa":
                            break
                        case "classtable":
                            break
                        default:
                            break
                        }
                    }
                }
            }
            
            
            return true
        } else {
            return false
        }
    }
    
}
